segments/weather.sh

```diff
@@ -63,8 +68,11 @@ __process_settings() {
 		export TMUX_POWERLINE_SEG_WEATHER_GREP="${TMUX_POWERLINE_SEG_WEATHER_GREP_DEFAULT}"
 	fi
 	if [ -z "$TMUX_POWERLINE_SEG_WEATHER_LOCATION" ]; then
-		echo "No weather location specified.";
-		exit 8
+		if [ -z "$TMUX_POWERLINE_SEG_WEATHER_LOCATION_DEFAULT" ]; then
```

Comment:
and revert this section to as it was before, because  if `TMUX_POWERLINE_SEG_WEATHER_LOCATION` is not set, there is nothing this segment can do => abort with error code about configuration problems.

Reply:
Thanks for your comments. 😃

This PR adds support for setting weather location in tmux-powerlinerc

- TMUX_POWERLINE_SEG_WEATHER_LOCATION is a variable to set in tmux-powerlinerc.
- TMUX_POWERLINE_SEG_WEATHER_LOCATION_DEFAULT is the variable you set in weather.sh.

# before

The setting method so far.

## If `TMUX_POWERLINE_SEG_WEATHER_LOCATION` of weather.sh is not set

weather.sh
```bash
# Prints the current weather in Celsius, Fahrenheits or lord Kelvins. The forecast is cached and updated with a period of $update_period.

# The update period in seconds.
update_period=600

TMUX_POWERLINE_SEG_WEATHER_DATA_PROVIDER_DEFAULT="yahoo"
TMUX_POWERLINE_SEG_WEATHER_UNIT_DEFAULT="c"
TMUX_POWERLINE_SEG_WEATHER_UPDATE_PERIOD_DEFAULT="600"

# Your location. Find a code that works for you:
# 1. Go to Yahoo weather http://weather.yahoo.com/
# 2. Find the weather for you location
# 3. Copy the last numbers in that URL. e.g. "http://weather.yahoo.com/united-states/california/newport-beach-12796587/" has the numbers "12796587"
TMUX_POWERLINE_SEG_WEATHER_LOCATION=""
```

"No weather location specified" will be displayed!

![image](https://user-images.githubusercontent.com/3356758/38972134-c6ba7b2e-43d9-11e8-854b-ca7c830cb604.png)

## When weather.sh's TMUX_POWERLINE_SEG_WEATHER_LOCATION is set

weather.sh
```bash
# Prints the current weather in Celsius, Fahrenheits or lord Kelvins. The forecast is cached and updated with a period of $update_period.

# The update period in seconds.
update_period=600

TMUX_POWERLINE_SEG_WEATHER_DATA_PROVIDER_DEFAULT="yahoo"
TMUX_POWERLINE_SEG_WEATHER_UNIT_DEFAULT="c"
TMUX_POWERLINE_SEG_WEATHER_UPDATE_PERIOD_DEFAULT="600"

# Your location. Find a code that works for you:
# 1. Go to Yahoo weather http://weather.yahoo.com/
# 2. Find the weather for you location
# 3. Copy the last numbers in that URL. e.g. "http://weather.yahoo.com/united-states/california/newport-beach-12796587/" has the numbers "12796587"
TMUX_POWERLINE_SEG_WEATHER_LOCATION="26237038"
```

![image](https://user-images.githubusercontent.com/3356758/38972381-106ff40a-43db-11e8-9109-341d4e7be718.png)

# after

## When set with tmux-powerlinerc

- Set the value to TMUX_POWERLINE_SEG_WEATHER_LOCATION.
- TMUX_POWERLINE_SEG_WEATHER_LOCATION_DEAFAULT is empty.

tmux-powerlinerc
```bash
# weather.sh {
        # The data provider to use. Currently only "yahoo" is supported.
        export TMUX_POWERLINE_SEG_WEATHER_DATA_PROVIDER="yahoo"
        # What unit to use. Can be any of {c,f,k}.
        export TMUX_POWERLINE_SEG_WEATHER_UNIT="c"
        # How often to update the weather in seconds.
        export TMUX_POWERLINE_SEG_WEATHER_UPDATE_PERIOD="600"
        # Name of GNU grep binary if in PATH, or path to it.
        export TMUX_POWERLINE_SEG_WEATHER_GREP="grep"
        # Your location. Find a code that works for you:
        # 1. Go to Yahoo weather http://weather.yahoo.com/
        # 2. Find the weather for you location
        # 3. Copy the last numbers in that URL. e.g. "http://weather.yahoo.com/united-states/california/newport-beach-12796587/" has the numbers "12796587"
        TMUX_POWERLINE_SEG_WEATHER_LOCATION="26237038"
# }
```

weather.sh
```bash
# Prints the current weather in Celsius, Fahrenheits or lord Kelvins. The forecast is cached and updated with a period of $update_period.

# The update period in seconds.
update_period=600

TMUX_POWERLINE_SEG_WEATHER_DATA_PROVIDER_DEFAULT="yahoo"
TMUX_POWERLINE_SEG_WEATHER_UNIT_DEFAULT="c"
TMUX_POWERLINE_SEG_WEATHER_UPDATE_PERIOD_DEFAULT="600"

# Your location. Find a code that works for you:
# 1. Go to Yahoo weather http://weather.yahoo.com/
# 2. Find the weather for you location
# 3. Copy the last numbers in that URL. e.g. "http://weather.yahoo.com/united-states/california/newport-beach-12796587/" has the numbers "12796587"
TMUX_POWERLINE_SEG_WEATHER_LOCATION_DEFAULT=""
```

Weather will be displayed!

![image](https://user-images.githubusercontent.com/3356758/38972381-106ff40a-43db-11e8-9109-341d4e7be718.png)

## When set with weather.sh

- TMUX_POWERLINE_SEG_WEATHER_LOCATION is empty.
- Set the value to TMUX_POWERLINE_SEG_WEATHER_LOCATION_DEAFAULT.

tmux-powerlinerc
```bash
# weather.sh {
        # The data provider to use. Currently only "yahoo" is supported.
        export TMUX_POWERLINE_SEG_WEATHER_DATA_PROVIDER="yahoo"
        # What unit to use. Can be any of {c,f,k}.
        export TMUX_POWERLINE_SEG_WEATHER_UNIT="c"
        # How often to update the weather in seconds.
        export TMUX_POWERLINE_SEG_WEATHER_UPDATE_PERIOD="600"
        # Name of GNU grep binary if in PATH, or path to it.
        export TMUX_POWERLINE_SEG_WEATHER_GREP="grep"
        # Your location. Find a code that works for you:
        # 1. Go to Yahoo weather http://weather.yahoo.com/
        # 2. Find the weather for you location
        # 3. Copy the last numbers in that URL. e.g. "http://weather.yahoo.com/united-states/california/newport-beach-12796587/" has the numbers "12796587"
        TMUX_POWERLINE_SEG_WEATHER_LOCATION=""
# }
```

weather.sh
```bash
# Prints the current weather in Celsius, Fahrenheits or lord Kelvins. The forecast is cached and updated with a period of $update_period.

# The update period in seconds.
update_period=600

TMUX_POWERLINE_SEG_WEATHER_DATA_PROVIDER_DEFAULT="yahoo"
TMUX_POWERLINE_SEG_WEATHER_UNIT_DEFAULT="c"
TMUX_POWERLINE_SEG_WEATHER_UPDATE_PERIOD_DEFAULT="600"

# Your location. Find a code that works for you:
# 1. Go to Yahoo weather http://weather.yahoo.com/
# 2. Find the weather for you location
# 3. Copy the last numbers in that URL. e.g. "http://weather.yahoo.com/united-states/california/newport-beach-12796587/" has the numbers "12796587"
TMUX_POWERLINE_SEG_WEATHER_LOCATION_DEFAULT="26237038"
```

Weather will be displayed!

![image](https://user-images.githubusercontent.com/3356758/38972381-106ff40a-43db-11e8-9109-341d4e7be718.png)

## When both are empty

- TMUX_POWERLINE_SEG_WEATHER_LOCATION is empty.
- TMUX_POWERLINE_SEG_WEATHER_LOCATION_DEAFAULT is empty.

tmux-powerlinerc
```bash
# weather.sh {
        # The data provider to use. Currently only "yahoo" is supported.
        export TMUX_POWERLINE_SEG_WEATHER_DATA_PROVIDER="yahoo"
        # What unit to use. Can be any of {c,f,k}.
        export TMUX_POWERLINE_SEG_WEATHER_UNIT="c"
        # How often to update the weather in seconds.
        export TMUX_POWERLINE_SEG_WEATHER_UPDATE_PERIOD="600"
        # Name of GNU grep binary if in PATH, or path to it.
        export TMUX_POWERLINE_SEG_WEATHER_GREP="grep"
        # Your location. Find a code that works for you:
        # 1. Go to Yahoo weather http://weather.yahoo.com/
        # 2. Find the weather for you location
        # 3. Copy the last numbers in that URL. e.g. "http://weather.yahoo.com/united-states/california/newport-beach-12796587/" has the numbers "12796587"
        TMUX_POWERLINE_SEG_WEATHER_LOCATION=""
# }
```

weather.sh
```bash
# Prints the current weather in Celsius, Fahrenheits or lord Kelvins. The forecast is cached and updated with a period of $update_period.

# The update period in seconds.
update_period=600

TMUX_POWERLINE_SEG_WEATHER_DATA_PROVIDER_DEFAULT="yahoo"
TMUX_POWERLINE_SEG_WEATHER_UNIT_DEFAULT="c"
TMUX_POWERLINE_SEG_WEATHER_UPDATE_PERIOD_DEFAULT="600"

# Your location. Find a code that works for you:
# 1. Go to Yahoo weather http://weather.yahoo.com/
# 2. Find the weather for you location
# 3. Copy the last numbers in that URL. e.g. "http://weather.yahoo.com/united-states/california/newport-beach-12796587/" has the numbers "12796587"
TMUX_POWERLINE_SEG_WEATHER_LOCATION_DEFAULT=""
```

"No weather location specified" will be displayed!

![image](https://user-images.githubusercontent.com/3356758/38972134-c6ba7b2e-43d9-11e8-854b-ca7c830cb604.png)